hail/src/main/scala/is/hail/expr/ir/EmitClassBuilder.scala

```diff
@@ -1055,6 +1052,24 @@ class EmitMethodBuilder[C](
       res.code
     })
   }
+
+  def implementLabel(label: CodeLabel)(f: EmitCodeBuilder => Unit): Unit = {
+    EmitCodeBuilder.scopedVoid(this) { cb =>
+      cb.define(label)
+      f(cb)
+//      assert(!cb.isOpenEnded)
```

Comment:
FYI

Reply:
These are assertions I would *like* to add here, but they fail in many of the usages in stream code. I think patterns like:
```
cb.ifx(b, 
 cb.goto(L1),
 cb.goto(L2))
  ```
leave the codebuilder open ended. I'll add a comment that these are aspirational assertions that should be reenabled when we fix the codebuilder state.